src/nemos/glm.py

```diff
@@ -120,7 +119,15 @@ def regularizer(self, regularizer: reg.Regularizer):
                 "The provided `solver` cannot be instantiated on "
                 "the GLM log-likelihood."
             )
-        self._regularizer = regularizer
+
+    @property
+    def solver(self) -> str:
+        return self._solver
+
+    @solver.setter
+    def solver(self, solver: str):
+        # check if solver/regularizer pairing valid
+        self._parse_regularizer_optimizer_params(regularizer=self.regularizer, solver=solver)
```

Comment:
```suggestion
        self._solver = self._parse_regularizer_optimizer_params(regularizer=self.regularizer, solver=solver)
```
Also this one may belong to BaseRegressor?

Again, why did we need the _solver here? where is it used?